ios/CodePush/CodePushDownloadHandler.m

```diff
@@ -22,12 +24,12 @@ -(void)download:(NSString*)url {
     NSURLRequest *request = [NSURLRequest requestWithURL:[NSURL URLWithString:url]
                                              cachePolicy:NSURLRequestUseProtocolCachePolicy
                                          timeoutInterval:60.0];
-    
     NSURLConnection *connection = [[NSURLConnection alloc] initWithRequest:request
                                                                   delegate:self
                                                           startImmediately:NO];
-    [connection scheduleInRunLoop:[NSRunLoop mainRunLoop]
-                          forMode:NSDefaultRunLoopMode];
+    NSOperationQueue *delegateQueue = [NSOperationQueue new];
+    delegateQueue.underlyingQueue = self.usingQueue;
+    [connection setDelegateQueue:delegateQueue];
```

Comment:
Now that we're doing this, I wonder if we still need to call `dispatch_async` within the `CodePush` class.